sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -4309,6 +4362,98 @@ protected internal virtual BlobContainerClient GetParentBlobContainerClientCore(
             return _parentBlobContainerClient;
         }
         #endregion
+
+        #region GetVersions
+        /// <summary>
+        /// Gets the Blob Versions associated with the Blob.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// An <see cref="Pageable{T}"/> of <see cref="BlobItem"/>
+        /// describing the versions assocaited with the blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Pageable<BlobItem> GetBlobVersions(CancellationToken cancellationToken = default)
+            => ContainerClient.GetBlobs(
+                states: BlobStates.Version,
+                prefix: Name,
```

Comment:
If I have blobs `foo` and `foobar` - listing versions for `foo` will include `foobar`. Should we filter them out before returning?

Reply:
In that case, I think we should cut this feature. If the customer is using a directory structure, and they have a blob named say “home”, which they also happen to be using as the name of their root virtual directory, this call could result in a very large number of blobs we have to filter though. I think the functionality I was trying to create with the SDK requires backend support.